src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs

```diff
@@ -98,18 +98,68 @@ private State()
 
             private bool ClashesWithExistingConstructor(TService service, SemanticDocument document, CancellationToken cancellationToken)
             {
-                var parameters = this.ParameterTypes.Zip(this.ParameterRefKinds, (t, r) => CodeGenerationSymbolFactory.CreateParameterSymbol(
-                    attributes: null,
-                    refKind: r,
-                    isParams: false,
-                    type: t,
-                    name: string.Empty)).ToList();
+                if (this.ParameterTypes == null ||
+                    this.ParameterRefKinds == null)
+                {
+                    return false;
+                }
+
+                int shortestLength = this.ParameterTypes.Count < this.ParameterRefKinds.Count ?
```

Comment:
previous implementation would have masked this issue using Zip, I'll change the semantics to return false if the lengths don't match